common/s2n-codec/src/encoder/estimator.rs

```diff
@@ -58,3 +61,50 @@ impl Encoder for EncoderLenEstimator {
         self.len
     }
 }
+/// The following harnesses were automatically generated by an
+/// experimental tool developed by the Kani team.
+#[cfg(test)]
+mod bolero_harnesses {
+    use super::*;
+    #[test]
+    #[cfg_attr(kani, kani::proof)]
+    fn bolero_test_27_new() {
+        bolero::check!()
+            .with_type()
+            .cloned()
+            .for_each(|capacity: usize| Some(EncoderLenEstimator::new(capacity)));
+    }
+    #[test]
+    #[cfg_attr(kani, kani::proof)]
+    fn bolero_test_28_overflowed() {
+        bolero::check!()
+            .with_type()
+            .cloned()
+            .for_each(|callee: EncoderLenEstimator| Some(callee.overflowed()));
+    }
+    #[test]
+    #[cfg_attr(kani, kani::proof)]
+    fn bolero_test_29_write_repeated() {
+        bolero::check!().with_type().cloned().for_each(
+            |(mut callee, count, _value): (EncoderLenEstimator, usize, u8)| {
+                Some(callee.write_repeated(count, _value))
+            },
```

Comment:
I guess you're pulling the variable names from the function declaration but I wonder if we could automatically remove the `_` prefix? Not a huge deal, just a minor nit.

Reply:
Also I'm assuming this panics when `callee.len + count` exceeds `usize::MAX`. With our current implementation, the Encoder impls are expected to panic on extreme values, since they don't return `Result` anywhere. Not sure what's the best way to capture that - do we filter out inputs in the harness that we expect to panic?

Reply:
Not exactly sure how to filter out the inputs. The code below still errors out with the input: 
```
Input: 
(
    EncoderLenEstimator {
        capacity: 0,
        len: 9151314442816847872,
    },
    9295429630892703744,
    0,
)1
``` 
which I figured would be filtered by the harness.
```rust
#[test]
    #[cfg_attr(kani, kani::proof)]
    fn bolero_test_29_write_repeated() {
        bolero::check!().with_type().cloned().for_each(
            |(mut callee, count, value): (EncoderLenEstimator, usize, u8)| {
                if callee.len + count > usize::MAX {
                  None
                } else {
                  Some(callee.write_repeated(count, value))
                }
            },
        );
    }
```

Reply:
Returning `None` from `for_each` will result in the harness failing. You need to use [filter](https://docs.rs/bolero/latest/bolero/struct.TestTarget.html#method.filter) instead.

Reply:
See [`impl IntoResult for Option<T>`](https://docs.rs/bolero-engine/latest/src/bolero_engine/result.rs.html#36-44)

Reply:
Tried to add a filter like this: 

```rust
fn encoder_len_estimator_write_repeated() {
        bolero::check!()
            .with_type()
            .cloned()
            .filter(|(callee, count, _): &(EncoderLenEstimator, usize, u8)| {
                (callee.len + count) > usize::MAX
            })
            .for_each(
                |(mut callee, count, value): (EncoderLenEstimator, usize, u8)| {
                    Some(callee.write_repeated(count, value))
                },
            );
    }
```

I still get an error in this test but now bolero isn't giving an input that causes the error. Instead I get this: 

```
---- encoder::estimator::tests::encoder_len_estimator_write_repeated stdout ----
thread 'encoder::estimator::tests::encoder_len_estimator_write_repeated' panicked at 'attempt to add with overflow', /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/core/src/ops/arith.rs:109:1
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    encoder::estimator::tests::encoder_len_estimator_write_repeated

test result: FAILED. 79 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 7.01s
```

Reply:
Oh that filter fails because it overflows in the filter check... This filter should get around that: 

```rust 
filter(|(callee, count, _): &(EncoderLenEstimator, usize, u8)| {
 count <= &(usize::MAX - callee.len)
})
```